lib/rack_request_parser.rb

```diff
@@ -0,0 +1,49 @@
+class RackRequestParser
```

Comment:
Does this need to take the `env`? It looks like it only needs the `request` to do its magic. This also reduces the required changes (not that there are many :))

Also, somewhat of a style concern, but Functional Dave wonders if this should just be the `values_to_check` method that takes the request as a parameter and spits out the values since its not doing anything other than reading the encapsulated state once.

Reply:
Yes, it needs `env`. If you look at the definition of `request`, it's an instance of `Rack::Request.new(env)`. Alternatively, I can pass in the request instance from the sanitizer into this class.